Modix/Modules/NoteModule.cs

```diff
@@ -0,0 +1,200 @@
+    public class NoteModule : ModuleBase
+    {
+        private readonly INoteCreatorService _noteCreatorService;
+        private readonly INoteRemoverService _noteRemoverService;
+        private readonly INoteRetrieverService _noteRetrieverService;
+
+        public NoteModule(INoteCreatorService noteCreatorService, INoteRemoverService noteRemoverService, INoteRetrieverService noteRetrieverService)
+        {
+            _noteCreatorService = noteCreatorService;
+            _noteRemoverService = noteRemoverService;
+            _noteRetrieverService = noteRetrieverService;
+        }
+
+
+        [Command("note add"), Summary("Adds a note to a user")]
+        public async Task CreateNote(IGuildUser user, string message)
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(message))
+                {
+                    Log.Warning($"User: {Context.Message.Author.Username} has tried to create a note without a message");
+                    await ReplyAsync("Please ensure that you have included a message with your note");
+                }
+
+                var result = await _noteCreatorService.Create(user, message, Context.Message.Author.Username);
+
+                if (!result)
+                {
+                    await ReplyAsync($"No note has been added for user {user.Username} due to an error. Please see the error logs");
+                }
+
+                await ReplyAsync($"Note has been created for user: {user.Username}");
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"NoteModule CreateNote failed with the following user: {user.Username} and message: {message}");
+                await ReplyAsync("Uh oh. Looks like you are using the command wrong or an error has occured.");
+            }
+        }
+
+        [Command("note remove"), Summary("Removes a note from a user")]
+        public async Task RemoveNote(int noteId)
+        {
+            try
+            {
+                var note = _noteRetrieverService.FindNoteById(noteId);
+
+                if (note == null)
+                {
+                    Log.Warning($"User: {Context.Client.CurrentUser} has tried to remove a note that does not exist");
+                    await ReplyAsync("Note does not exist for that Id");
+                }
+
+                var result = await _noteRemoverService.Remove(note);
+
+                if (!result)
+                {
+                    await ReplyAsync($"Error occurred and note could not be removed");
+                }
+
+                await ReplyAsync($"Note {noteId} has been removed");
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"NoteModule RemoveNote failed with the following noteId: {noteId}");
+                await ReplyAsync($"Failed to remove the note: {noteId}");
```

Comment:
This catch block would also be entered if one of the calls to `ReplyAsync` throws (in which case, it will _probably_ throw again here). It may be worth reducing the scope of the `try` block to only cover the operations involving `_noteRemoverService` (same for the other commands in this class, too).